src/adapters/github.js

```diff
@@ -118,12 +118,17 @@ class GitHub extends PjaxAdapter {
     const pullNumber = isPR && showOnlyChangedInPR ? typeId : null
 
     // Skip non-code page unless showInNonCodePage is true
-    if (!showInNonCodePage && type && !~['tree', 'blob'].indexOf(type)) {
+    const isCodePage = ['tree', 'blob', 'commit'].indexOf(type) >= 0
+    if (!showInNonCodePage && type && !isCodePage) {
       return cb()
     }
 
     // Get branch by inspecting page, quite fragile so provide multiple fallbacks
+    const isBranchOrSha = isCodePage && typeId
+    const branchOrSha = isBranchOrSha && typeId ? typeId : null
```

Comment:
The second `typeId` check isn't necessary. Should just merge these 2 lines to: 
```
const branchOrSha = isCodePage && typeId
```
But really, that's the whole point of the "branch" variable. Might just combine:

```
const branch = 
   // Branch/tag/commit from URL
   (isCodePage && typeId) ||
   // Code page DOM
   // .. the rest
```